File: src/main/java/kr/co/dealmungchi/hotdealbatch/domain/service/HotdealService.java
package kr.co.dealmungchi.hotdealbatch.domain.service;

import org.springframework.stereotype.Service;

import kr.co.dealmungchi.hotdealbatch.service.filehandler.FileHandler;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

/**
 * Service for managing hot deal-related operations.
 * Currently handles thumbnail uploads.
 */
@Slf4j
@Service
@RequiredArgsConstructor
public class HotdealService {

    private final FileHandler fileHandler;
    
    /**
     * Uploads a thumbnail image and returns its path.
     *
     * @param hash The hash identifier for the thumbnail, used as the filename
     * @param data The binary data of the thumbnail image
     * @return The path where the thumbnail was uploaded
     */
    public String uploadThumbnail(String hash, byte[] data) {
        String path = fileHandler.upload(hash, data);
        log.debug("Thumbnail uploaded to path: {}", path);
        return path;
    }
}

File: src/main/kotlin/kr/co/dealmungchi/hotdealbatch/domain/service/HotdealService.kt
package kr.co.dealmungchi.hotdealbatch.domain.service

import kr.co.dealmungchi.hotdealbatch.service.filehandler.FileHandler
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

/**
 * Service for managing hot deal-related operations.
 * Currently handles thumbnail uploads.
 */
@Service
class HotdealService(private val fileHandler: FileHandler) {
    
    private val log = LoggerFactory.getLogger(HotdealService::class.java)
    
    /**
     * Uploads a thumbnail image and returns its path.
     *
     * @param hash The hash identifier for the thumbnail, used as the filename
     * @param data The binary data of the thumbnail image
     * @return The path where the thumbnail was uploaded
     */
    fun uploadThumbnail(hash: String, data: ByteArray): String {
        return fileHandler.upload(hash, data)
    }
    
    /**
     * Uploads a thumbnail image and returns its path as a Mono.
     * Reactive version of uploadThumbnail.
     *
     * @param hash The hash identifier for the thumbnail, used as the filename
     * @param data The binary data of the thumbnail image
     * @return A Mono that resolves to the path where the thumbnail was uploaded
     */
    fun uploadThumbnailReactive(hash: String, data: ByteArray): Mono<String> {
        return Mono.fromCallable { uploadThumbnail(hash, data) }
    }
}